packages/block-library/src/navigation/edit/index.js

```diff
@@ -73,6 +73,30 @@ import AccessibleDescription from './accessible-description';
 import AccessibleMenuDescription from './accessible-menu-description';
 import { unlock } from '../../lock-unlock';
 
+// Custom hook for managing responsive menu state.
+function useResponsiveMenu( navRef ) {
+	const [ isResponsiveMenuOpen, setResponsiveMenuVisibility ] =
+		useState( false );
+
+	useEffect( () => {
+		if ( ! navRef.current ) {
+			return;
+		}
+
+		const htmlElement = navRef.current.closest( 'html' );
+
+		// Add a `has-modal-open` class to the <html> when the responsive
+		// menu is open. This reproduces the same behavior of the frontend.
```

Comment:
It does feel a bit odd for a block to be manipulating the root node but if this brings it inline with the frontend perhaps it's already been decided this is ok?

Reply:
This does feel unusual. Is there any precedent?

Reply:
Is this actually needed? I used the testing instructions against `trunk` and was able to replicate the issue but only on the front of the site and not in the Editor.


https://github.com/user-attachments/assets/eb0ef33e-8641-4565-ae07-754052df9dcc

Reply:
Unfortunately, it's really needed and also happens in the editor.

<img width="496" alt="Screenshot 2024-10-15 at 09 51 06" src="https://github.com/user-attachments/assets/7a059904-124d-4cd8-ae94-41c35dcc1099">

Since it's the behavior in the frontend, I think it would be fine to have the same behavior replicated in the editor.

> I used the testing instructions against trunk and was able to replicate the issue but only on the front of the site and not in the Editor.

I noticed sometimes some kind of cache. I didn't dig into to figure out, but sometimes when it delayed I restarted the build and restarted the browser, also marking the "Disable cache" in my network tab.

Reply:
Are there specific steps to replicate that state? I tried following the PR instructions but it didn't manifest. Maybe it was prior to the rebase?

Reply:
Ah, sorry! I forgot to add to the instructions to apply the https://github.com/WordPress/gutenberg/pull/66074 before testing.

After the rebase, if you comment the style change, you will see the error happening.

Reply:
> This does feel unusual. Is there any precedent?

I agree it feels unusual and I couldn't find a precedent with other core blocks. That said, in this case one of the things I kind of like about adding and removing classes via the `useEffect` and attaching to the node, is that the logic is contained within the block itself, and the editor doesn't need to be aware of it — so, we're not handling the state of the modal within the block-editor state for example.

Reply:
One downside is that this sort of makes the Navigation block dependent on the Cover block adjusting its own z-index values. It's not so much that the coupling exists but that it is rather hidden to anyone working on the blocks in the future (hence the importance of the e2e test I guess).

With a lack of better alternatives, I'd be on board with progressing this fix and iterating again if needed.

Reply:
> With a lack of better alternatives, I'd be on board with progressing this fix and iterating again if needed.

I was just thinking the same. +1 from me for progressing with this and revisiting further down the track.

Reply:
> The problem would be the block deprecation

I'm missing what the problem with a block deprecation would be. Can you expand on this aspect?

> a complex code to maintain both versions to not break the styles

We unfortunately have to keep a fair bit of CSS around for deprecated versions of blocks. I'm guessing what would be required here isn't a lot.

As for the block's render callback, it doesn't look overly complex at a glance. So it could likely absorb any slight change in logic for BC.

Reply:
Hi @aaronrobertshaw! I could be missing something, but I think for deprecations changing the markup it looks "broken" (with the warning) in the editor and the user needs to click in a button to fix the block, right?

So I thought that it would be better to avoid it if we could for the cover block because is very common and it is a wrapper component.

For the render callback, I'm not sure if I understand what you mean. Do you mean to invert the position of the elements dynamically only? If so, I'm not sure if it's a good idea because we need the same in the `edit`. Unless we change the order in the `edit` for the editor, and change in the render callback for the frontend, although I think it would be a weird code for maintenance.

Reply:
I just checked I was wrong about the deprecation and it's transparent for the user. I'll raise another PR to improve it more. And the solution of the `has-modal-open` will be only for backward compatibility. =)

Reply:
I raised [this PR](https://github.com/WordPress/gutenberg/pull/66249). I'd appreciate if someone could review it to improve the solution.

I also tried to update the fixtures through `npm run fixtures:regenerate`, but I received some errors. If someone could explain me what is happening and how I could fix, it would also be very helpful to save some time. =)

Reply:
> I just checked I was wrong about the deprecation and it's transparent for the user

Yep, the only real sign within the editor should be a console message stating the block was migrated.

> And the solution of the has-modal-open will be only for backward compatibility

This PR says it should be in the 19.6 release. So if we do go with the alternative solution I don't think there'll be any need to keep the `.has-modal-open` class styles around for BC. The change introducing it won't have been release in other words.

I'll comment to that effect when I get to review the alternate PR.

> I also tried to update the fixtures through npm run fixtures:regenerate, but I received some errors.

Without looking I can't say if this is the reason or not but another misconception about deprecations is that they are chained together to migrate one deprecated version to the next. All the way to the latest version.

Instead, all deprecations need to be updated (if required) to migrate that deprecated version to the latest. We can continue the discussion over on https://github.com/WordPress/gutenberg/pull/66249 as needed.

Thanks for spinning it up 🚀

Reply:
> This PR says it should be in the 19.6 release. So if we do go with the alternative solution soon I don't think there'll be any need to keep the .has-modal-open class styles around for BC. The change introducing it won't have been released in other words.

It's good to have that anyway, so it's fixed everywhere not needing a manual save in all the broken posts/templates. :)

Reply:
Wasn't the original issue only in the editor? If that's the case, the deprecation will migrate and fix that display without needing further styles.